lib/ReactViews/ActionBar/ActionBarPortal.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import styled from "styled-components";
+import ViewState from "../../ReactViewModels/ViewState";
+import { Portal } from "../StandardUserInterface/Portal";
+import { withViewState } from "../StandardUserInterface/ViewStateContext";
+
+export const ActionBarPortalId = "action-bar-ui-portal";
+
+interface PropsType {
+  viewState: ViewState;
+
+  /**
+   * Toggles portal visibility.
+   *
+   * This is used for CSS transition animation.
+   */
+  show: boolean;
+}
+
+/**
+ * A Portal to show ActionBar UI.
+ */
+const ActionBarPortal: React.FC<PropsType> = ({ viewState, show }) => {
+  return <Container viewState={viewState} id={ActionBarPortalId} show={show} />;
+};
+
+const Container = styled(Portal)<{ show: boolean }>`
+  display: flex;
+  position: absolute;
+  height: 56px;
+  visibility: ${(p) => (p.show ? "visible" : "hidden")};
+  max-width: 60%;
+  bottom: ${(p) => (p.show ? "80px" : "-56px")};
+  left: 0;
+  right: 0;
+  margin: auto;
+
+  /* Animate slide in-out */
+  transition: all 0.2s;
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+`;
+
+export default withViewState(ActionBarPortal);
```

Comment:
Prefer use of `useViewState` and reduce passing viewState through props to reduce noise. Could you update StandardUserInterface/Portal to use `useViewState`? Then I think you won't need viewState here.